providers/packetframe/packetframeProvider.go

```diff
@@ -0,0 +1,245 @@
+package packetframe
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/StackExchange/dnscontrol/v3/models"
+	"github.com/StackExchange/dnscontrol/v3/pkg/diff"
+	"github.com/StackExchange/dnscontrol/v3/providers"
+)
+
+// packetframeProvider is the handle for this provider.
+type packetframeProvider struct {
+	client      *http.Client
+	baseURL     *url.URL
+	token       string
+	domainIndex map[string]zone
+}
+
+var defaultNameServerNames = []string{
+	"ns1v4.packetframe.com",
+	"ns2v4.packetframe.com",
+}
+
+// NewPacketframe creates the provider.
+func NewPacketframe(m map[string]string, metadata json.RawMessage) (providers.DNSServiceProvider, error) {
+	if m["apikey"] == "" {
+		return nil, fmt.Errorf("missing Packetframe token")
+	}
+
+	baseURL, err := url.Parse(defaultBaseURL)
+	if err != nil {
+		return nil, fmt.Errorf("invalid base URL for Packetframe")
+	}
+	client := http.Client{}
+
+	api := &packetframeProvider{client: &client, baseURL: baseURL, token: m["apikey"]}
+
+	// Get a domain to validate the token
```

Comment:
An initializer such as `NewPacketframe` should not generate network traffic.  Can this validation be done on first use instead?